trillian/docs/ManualDeployment.md

```diff
@@ -0,0 +1,460 @@
+`trillian_log_signer` should include a mention of the new tree.
+```
+I1011 16:44:16.160069  176101 log_operation_manager.go:210] create master election goroutine for 2385931157013381257
+I1011 16:44:17.160875  176101 log_operation_manager.go:246] now acting as master for 1 / 1, master for: <log-2385931157013381257>
+```
+
+
+## CT Personality
+
+Trillian provides a general gRPC API for Merkle tree operations, but relies on a
+*personality* to perform operations that are specific to the particular
+[transparency application](https://github.com/google/trillian/blob/master/docs/TransparentLogging.md).
+
+For Certificate Transparency, the `ctfe/` directory holds a Trillian personality
+which:
+ - provides the HTTP/JSON API entrypoints described by [RFC 6962](https://tools.ietf.org/html/rfc6962)
+ - checks that submissions to the Log are valid X.509 certificates, with a
+   chain of signatures that reaches an acceptable root.
+
+The CTFE personality is generally stateless, and is controlled by a
+[configuration file](#ctfe-configuration); the following subsections describe
+the key components of this file.
+
+### Key Generation
+
+Each CT Log needs to have a private key that is used to sign cryptographic
+content from the Log.  The [OpenSSL](https://www.openssl.org/) command line can
+be used to
+[generate](https://wiki.openssl.org/index.php/Command_Line_Elliptic_Curve_Operations#Generating_EC_Keys_and_Parameters)
+a suitable private key.
+
+```bash
+% openssl ecparam -name prime256v1 > privkey.pem # generate parameters file
+% openssl ecparam -in privkey.pem -genkey -noout >> privkey.pem # generate and append private key
+% openssl ec -in privkey.pem -pubout -out pubkey.pem # generate corresponding public key
+```
+
+The private key must either be for elliptic curves using NIST P-256 (as shown
+here), or for RSA signatures with SHA-256 and a 2048 bit (or larger) key
+([RFC 6962 s2.1.4](https://tools.ietf.org/html/rfc6962#section-2.1.4)).
+
+**Cross-check**: Confirm that the key is well-formed and readable:
+```bash
+% openssl ec -in privkey.pem -noout -text # check key is readable
+read EC key
+Private-Key: (256 bit)
+priv:
+    00:b5:99:8c:7b:f2:5b:0c:a1:3a:26:b0:12:e2:b7:
+    dd:c6:89:a6:49:3c:1d:26:70:44:ad:4a:34:91:2d:
+    b6:33:a3
+pub:
+    04:16:44:9b:04:47:ae:93:f4:14:94:7b:f7:ba:ae:
+    5e:6b:53:e3:b4:85:55:ab:f4:06:0f:65:36:bd:f7:
+    5f:d7:74:0c:e5:30:c6:a9:0e:0d:40:70:5d:b2:70:
+    92:cc:b9:bc:c7:d4:16:7e:96:24:52:6e:1a:a4:28:
+    43:d0:b5:97:72
+ASN1 OID: prime256v1
+NIST CURVE: P-256
+% openssl pkey -pubin -in pubkey.pem -text -noout
+Public-Key: (256 bit)
+pub:
+    04:16:44:9b:04:47:ae:93:f4:14:94:7b:f7:ba:ae:
+    5e:6b:53:e3:b4:85:55:ab:f4:06:0f:65:36:bd:f7:
+    5f:d7:74:0c:e5:30:c6:a9:0e:0d:40:70:5d:b2:70:
+    92:cc:b9:bc:c7:d4:16:7e:96:24:52:6e:1a:a4:28:
+    43:d0:b5:97:72
+ASN1 OID: prime256v1
+NIST CURVE: P-256
+```
+
+**Cross-check**: Once the CTFE is configured and running, the `ctclient`
```

Comment:
Maybe move this down after the discussion around configuring and running the ctfe?
Anybody trying this cross check now will fail, and for anyone paying closer attention `aramis` is unintroduced currently.

Reply:
Moving all the cross-checks after ctfe startup would end up duplicating the sub-section structure, so I prefer to leave these in-line. I've added `([below](#ctfe-start-up))` markers to try to reinforce the order though.